# Missing and Default Values

In [7]:
import pandas as pd
import seaborn as sns
import os

DATA_DIR = "../original_data"
TRAIN_VALUES_FILE = os.path.join(DATA_DIR, "train_values.csv")

sns.set_theme(style="whitegrid")
train_values = pd.read_csv(TRAIN_VALUES_FILE)

## Missing Values

In [8]:
train_values.isna().sum()

building_id                               0
geo_level_1_id                            0
geo_level_2_id                            0
geo_level_3_id                            0
count_floors_pre_eq                       0
age                                       0
area_percentage                           0
height_percentage                         0
land_surface_condition                    0
foundation_type                           0
roof_type                                 0
ground_floor_type                         0
other_floor_type                          0
position                                  0
plan_configuration                        0
has_superstructure_adobe_mud              0
has_superstructure_mud_mortar_stone       0
has_superstructure_stone_flag             0
has_superstructure_cement_mortar_stone    0
has_superstructure_mud_mortar_brick       0
has_superstructure_cement_mortar_brick    0
has_superstructure_timber                 0
has_superstructure_bamboo       

As we see (or don't see) in the output, there are no missing values in the data set.

## Default Values

### Age

In [9]:
train_values["age"].value_counts().sort_index(ascending=False)

age
995     1390
200      106
195        2
190        3
185        1
180        7
175        5
170        6
165        2
160        6
155        1
150      142
145        3
140        9
135        5
130        9
125       37
120      180
115       21
110      100
105       89
100     1364
95       414
90      1085
85       847
80      3055
75       512
70      1975
65      1123
60      3612
55      2033
50      7257
45      4711
40     10559
35     10710
30     18028
25     24366
20     32182
15     36010
10     38896
5      33697
0      26041
Name: count, dtype: int64

995 is most probably the default value for buildings with no age info. Other than that, it also seems that the number of buildings with ages 150 and 200 is suspiciously larger than their neighbors. That is why their ages are probably rounded to those numbers.